### PR TITLE
[Feature] Release Benchmark for SAGAN and SNGAN

### PR DESCRIPTION
**Pull request by LeoXing1996**
1. Release benchmark for SAGAN and SNGAN
2. Add dataset configs for extracting inception state.
3. Convert interpolation method from Bicubic to Bilinear in all ImageNet dataset configs.

### CODE REVIEW

File: configs/_base_/datasets/imagenet_inception_stat.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# https://github.com/ajbrock/BigGAN-PyTorch/blob/master/make_hdf5.py
+# https://github.com/ajbrock/BigGAN-PyTorch/blob/master/calculate_inception_moments.py  # noqa
+
+# Importantly, the `to_rgb` is set to `True` to convert image orders to RGB.
+img_norm_cfg = dict(
+    mean=[127.5, 127.5, 127.5], std=[127.5, 127.5, 127.5], to_rgb=True)
+train_pipeline = [
+    dict(type='LoadImageFromFile'),
+    dict(type='CenterCropLongEdge', keys=['img']),
+    dict(type='Resize', size=(128, 128), backend='pillow'),
```
Comment (nbei) [inline configs/_base_/datasets/imagenet_inception_stat.py:17]: You may rename this file to `imagenet_128x128_inception_stat.py`

File: configs/_base_/datasets/sagan_128x128.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# define GAN model
+model = dict(
```
Comment (nbei) [inline configs/_base_/datasets/sagan_128x128.py:2]: You should put this file to `configs/__base__/models`

File: configs/sagan/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+| sagan_32  | CIFAR10  |     w/o      |   5    |                       64x1x1                        |    500000    |        [ 9.3217 (400000) ](https://download.openmmlab.com/mmgen/sagan/sagan_cifar10_32_lr2e-4_ndisc5_b64x1_woReUinplace_is-iter400000_20210730_125743-4008a9ca.pth)         |        [ 9.4252 (480000) ](https://download.openmmlab.com/mmgen/sagan/sagan_cifar10_32_lr2e-4_ndisc5_b64x1_woReUinplace_fid-iter480000_20210730_125449-d50568a4.pth )        |        [config](https://github.com/open-mmlab/mmgeneration/tree/master/configs/sagan/sagan_32_woReLUinplace_cifar10_b64x1_lr-2e-4_ndisc5.py)        |       [Log](https://download.openmmlab.com/mmgen/sagan/sagan_cifar10_32_lr2e-4_ndisc5_b64x1_woReUinplace_20210730_125449_fid-d50568a4_is-04008a9ca.json)        |
```
Comment (nbei) [inline configs/sagan/README.md:23]: If `N_GRAD_ACCU` is always 1, we can just ignore it in this benchmark.

File: configs/sagan/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+| sagan_128 | ImageNet |     w/o      |   1    |                       64x4x1                        |   1000000    | [ 31.5938 (980000) ](https://download.openmmlab.com/mmgen/sagan/sagan_imagenet1k_128_Glr1e-4_Dlr4e-4_ndisc1_b32x4_woReLUinplace_is-iter980000_20210730_163140-cfbebfc6.pth) | [ 34.7838 (950000) ](https://download.openmmlab.com/mmgen/sagan/sagan_imagenet1k_128_Glr1e-4_Dlr4e-4_ndisc1_b32x4_woReLUinplace_fid-iter950000_20210730_163431-d7916963.pth) | [config](https://github.com/open-mmlab/mmgeneration/tree/master/configs/sagan/sagan_128_woReLUinplace_imagenet1k_b64x4_Glr-1e-4_Dlr-4e-4_ndisc1.py) | [Log](https://download.openmmlab.com/mmgen/sagan/sagan_imagenet1k_128_Glr1e-4_Dlr4e-4_ndisc1_b32x4_woReLUinplace_20210730_163431_fid-d7916963_is-cfbebfc6.json) |
+
+'\*' Iteration counting rule in our implementation is different from others. If you want to align with other codebases, you can use the following conversion formula:
+```
+Total_iters* = Total_iter / (n_disc * n_grad_accu)
```
Comment (nbei) [inline configs/sagan/README.md:29]: You should tell users what the meaning of `total_iters*` and `total_iters` is.

Total --> total

File: configs/sagan/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+```
+
+We also provide converted pretrain model from [Pytorch-StudioGAN](https://github.com/POSTECH-CVLab/PyTorch-StudioGAN).
```
Comment (nbei) [inline configs/sagan/README.md:32]: converted pre-trained models

File: configs/sagan/README.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+| :-------: | :------: | :----------: | :----: | :---------: | :---------: | :----: | :-----: | :----: | :-----: | :-------------------------------------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------: |
+| sagan_32  | CIFAR10  |      w       |   5    |      1      |   100000    | 9.116  | 10.2011 | 8.680  | 14.009  |   [Download](https://download.openmmlab.com/mmgen/sagan/sagan_32_cifar10_convert-studio-rgb_20210730_153321-080da7e2.pth)   | [Download](https://drive.google.com/drive/folders/1FA8hcz4MB8-hgTwLuDA0ZUfr8slud5P_) |
+| sagan_128 | ImageNet |      w       |   1    |      1      |   1000000   | 27.367 | 40.1162 | 29.848 | 34.726  | [Download](https://download.openmmlab.com/mmgen/sagan/sagan_128_imagenet1k_convert-studio-rgb_20210730_153357-eddb0d1d.pth) | [Download](https://drive.google.com/drive/folders/1ZYaqeeumDgxOPDhRR5QLeLFIpgBJ9S6B) |
+
+
+* '\*\*' denote results evaluated with our pipeline.
```
Comment (nbei) [inline configs/sagan/README.md:41]: `**` and `***` are very ugly. You can just use names with or without `*` to indicate the difference.

File: configs/sagan/README.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+* '\*\*\*' denote results released by Pytorch-StudioGAN.
+
+For IS evaluation,
```
Comment (nbei) [inline configs/sagan/README.md:44]: For IS metric, our implementation is different from PyTorch-Studio GAN in the following aspects:

File: configs/sagan/README.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+1. We use [Tero's Inception](https://nvlabs-fi-cdn.nvidia.com/stylegan2-ada-pytorch/pretrained/metrics/inception-2015-12-05.pt) for feature extraction.
+2. We use bicubic interpolation with PIL backend to resize image before feed them to Inception.
+
+For FID evaluation, difference between two repositories mainly on the selection of images for ground truth activation (inception state) calculation. In ours, we follow the pipeline of BigGAN for more convincing results. Detailly, we use the entire training set (50k and 1281167 images for CIFAR10 and ImageNet1k) without shuffle for GT inception state extracting. We also use [Tero's Inception](https://nvlabs-fi-cdn.nvidia.com/stylegan2-ada-pytorch/pretrained/metrics/inception-2015-12-05.pt) for feature extraction.
```
Comment (nbei) [inline configs/sagan/README.md:48]: For FID evaluation, differences between PyTorch Studio GAN and ours are mainly on the selection of real samples. In MMGen, we follow the pipeline of BigGAN, where the whole training set is adopted to extract inception statistics. Besides, we also use [Tero's Inception](https://nvlabs-fi-cdn.nvidia.com/stylegan2-ada-pytorch/pretrained/metrics/inception-2015-12-05.pt) for feature extraction.

File: configs/sagan/README.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+You can download the preprocessed inception state by the following url: [CIFAR10](https://download.openmmlab.com/mmgen/evaluation/fid_inception_pkl/cifar10.pkl) and [ImageNet1k](https://download.openmmlab.com/mmgen/evaluation/fid_inception_pkl/imagenet.pkl).
+
+You can use following scripts to extract those inception states by yourself.
```
Comment (nbei) [inline configs/sagan/README.md:52]: You can use the following commands to extract those inception states by yourself.

File: configs/sagan/sagan_32_wReLUinplace_cifar10_b64x1_lr-2e-4_ndisc5.py [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+lr_config = None
+checkpoint_config = dict(interval=10000, by_epoch=False, max_keep_ckpts=20)
+custom_hooks = [
+    dict(
+        type='VisualizeUnconditionalSamples',
+        output_dir='training_samples',
+        interval=1000)
+]
+
+inception_pkl = \
```
Comment (nbei) [inline configs/sagan/sagan_32_wReLUinplace_cifar10_b64x1_lr-2e-4_ndisc5.py:22]: remove `\`

File: configs/sngan_proj/README.md [hunk 11]
```diff
@@ -11,29 +11,38 @@
...
+| sngan_proj_128 | ImageNet |      w       |   2    |    2000000    |                                                                                Coming soon                                                                                |                                                                                     Comming soon                                                                                      |                                                                          Comming Soon                                                                          |                                                                                Comming Soon                                                                                |
 
+'\*' Iteration counting rule in our implementation is different from others. If you want to align with other codebases, you can use the following conversion formula:
+```
+Total_iters* = Total_iter / (n_disc * n_grad_accu)
```
Comment (nbei) [inline configs/sngan_proj/README.md:32]: You may revise this file according to the comments in the sagan.

File: configs/sagan/README.md (hunk 3 above)
Comment (LeoXing1996) [reply configs/sagan/README.md:23]: `N_GRAD_ACCU` is set for our ongoing setting, which `N_GRAD_ACCU=2`.

File: configs/sagan/README.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+| sagan_128 | ImageNet |     w/o      |   1    |                 64x4                 |   1000000    | [ 31.5938 (980000) ](https://download.openmmlab.com/mmgen/sagan/sagan_imagenet1k_128_Glr1e-4_Dlr4e-4_ndisc1_b32x4_woReLUinplace_is-iter980000_20210730_163140-cfbebfc6.pth) | [ 34.7838 (950000) ](https://download.openmmlab.com/mmgen/sagan/sagan_imagenet1k_128_Glr1e-4_Dlr4e-4_ndisc1_b32x4_woReLUinplace_fid-iter950000_20210730_163431-d7916963.pth) | [config](https://github.com/open-mmlab/mmgeneration/tree/master/configs/sagan/sagan_128_woReLUinplace_imagenet1k_b64x4_Glr-1e-4_Dlr-4e-4_ndisc1.py) | [Log](https://download.openmmlab.com/mmgen/sagan/sagan_imagenet1k_128_Glr1e-4_Dlr4e-4_ndisc1_b32x4_woReLUinplace_20210730_163431_fid-d7916963_is-cfbebfc6.json) |
+
+'\*' Iteration counting rule in our implementation is different from others. If you want to align with other codebases, you can use the following conversion formula:
+```
+MMGEN_TOTAL_ITERS = TOTAL_ITERS / n_disc
+```
+
```
Comment (nbei) [inline configs/sagan/README.md:31]: In fact, I am still confusing about `total_iters`. Do you mean

```
total_iters (biggan/pytorch studio gan) = our_totoal_iters / n_disc
```

File: configs/sagan/README.md [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+| :-------: | :------: | :----------: | :----: | :---------: | :----------------: | :-----------------: | :----------: | :-----------: | :-------------------------------------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------: |
+| sagan_32  | CIFAR10  |      w       |   5    |   100000    |       9.116        |       10.2011       |    8.680     |    14.009     |   [Download](https://download.openmmlab.com/mmgen/sagan/sagan_32_cifar10_convert-studio-rgb_20210730_153321-080da7e2.pth)   | [Download](https://drive.google.com/drive/folders/1FA8hcz4MB8-hgTwLuDA0ZUfr8slud5P_) |
+| sagan_128 | ImageNet |      w       |   1    |   1000000   |       27.367       |       40.1162       |    29.848    |    34.726     | [Download](https://download.openmmlab.com/mmgen/sagan/sagan_128_imagenet1k_convert-studio-rgb_20210730_153357-eddb0d1d.pth) | [Download](https://drive.google.com/drive/folders/1ZYaqeeumDgxOPDhRR5QLeLFIpgBJ9S6B) |
+
+
+* `Ours Pipeline` denote results evaluated with our pipeline.
```
Comment (nbei) [inline configs/sagan/README.md:41]: Ours ---> Our
denote ---> denotes

File: configs/sagan/README.md [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+* `Release` denote results released by Pytorch-StudioGAN.
```
Comment (nbei) [inline configs/sagan/README.md:42]: `Release` ---> `StudioGAN`

File: configs/sagan/README.md (hunk 8 above)
Comment (nbei) [reply configs/sagan/README.md:48]: **are** mainly on

Comment (nbei) [reply configs/sagan/README.md:48]: It seems that you have not modified the sentences.

File: configs/sngan_proj/README.md [hunk 15]
```diff
@@ -11,43 +11,54 @@
...
-|:--------------:|:----------:|:------:|:-------:|:------:|:-------:|-----------------------------------------------------------------------------------------------------------------------------:|:------------------------------------------------------------------------------------:|
-| sngan_proj_32  |  CIFAR10   | 9.372  |  9.205  | 8.677  | 13.248  |    [Download](https://download.openmmlab.com/mmgen/sngan_proj/sngan_cifar10_convert-studio-rgb_20210709_111346-2979202d.pth) | [Download](https://drive.google.com/drive/folders/16s5Cr-V-NlfLyy_uyXEkoNxLBt-8wYSM) |
-| sngan_proj_128 | ImageNet1k | 30.218 | 29.8199 | 32.247 | 26.792  | [Download](https://download.openmmlab.com/mmgen/sngan_proj/sngan_imagenet1k_convert-studio-rgb_20210709_111406-877b1130.pth) | [Download](https://drive.google.com/drive/folders/1Ek2wAMlxpajL_M8aub4DKQ9B313K8XhS) |
 
+* `Ours Pipeline` denote results evaluated with our pipeline.
+* `Release` denote results released by Pytorch-StudioGAN.
```
Comment (nbei) [inline configs/sngan_proj/README.md:46]: Revise according to the comments in sagan

File: configs/sagan/README.md (hunk 12 above)
Comment (LeoXing1996) [reply configs/sagan/README.md:31]: Oh, my fault.
This is a big mistake 😢

File: configs/sngan_proj/README.md [hunk 16]
```diff
@@ -11,43 +11,53 @@
...
+|     Models     | Dataset  | Inplace ReLU | n_disc | Total Iters | IS (Ours Pipeline) | FID (Ours Pipeline) | IS (StudioGAN) | FID (StudioGAN) |                                                           Download                                                           |                                Original Download link                                |
+| :------------: | :------: | :----------: | :----: | :---------: | :----------------: | :-----------------: | :----------: | :-----------: | :--------------------------------------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------: |
+| sngan_proj_32  | CIFAR10  |      w       |   5    |   100000    |       9.372        |        9.205        |    8.677     |    13.248     |  [Download](https://download.openmmlab.com/mmgen/sngan_proj/sngan_cifar10_convert-studio-rgb_20210709_111346-2979202d.pth)   | [Download](https://drive.google.com/drive/folders/16s5Cr-V-NlfLyy_uyXEkoNxLBt-8wYSM) |
```
Comment (LeoXing1996) [inline configs/sngan_proj/README.md:41]: CIFAR10 FID: 9.205 -> 10.2011

File: configs/sngan_proj/README.md [hunk 17]
```diff
@@ -11,43 +11,53 @@
...
-| sngan_proj_32  |  CIFAR10   | 9.372  |  9.205  | 8.677  | 13.248  |    [Download](https://download.openmmlab.com/mmgen/sngan_proj/sngan_cifar10_convert-studio-rgb_20210709_111346-2979202d.pth) | [Download](https://drive.google.com/drive/folders/16s5Cr-V-NlfLyy_uyXEkoNxLBt-8wYSM) |
-| sngan_proj_128 | ImageNet1k | 30.218 | 29.8199 | 32.247 | 26.792  | [Download](https://download.openmmlab.com/mmgen/sngan_proj/sngan_imagenet1k_convert-studio-rgb_20210709_111406-877b1130.pth) | [Download](https://drive.google.com/drive/folders/1Ek2wAMlxpajL_M8aub4DKQ9B313K8XhS) |
+|     Models     | Dataset  | Inplace ReLU | n_disc | Total Iters | IS (Ours Pipeline) | FID (Ours Pipeline) | IS (StudioGAN) | FID (StudioGAN) |                                                           Download                                                           |                                Original Download link                                |
```
Comment (LeoXing1996) [inline configs/sngan_proj/README.md:39]: Ours Pipeline -> Our Pipeline
